### PR TITLE
frontend: allow admins to change a users role

### PR DESCRIPTION
**Pull request by flosse**

### CODE REVIEW

File: src/core/usecases/archive_comments.rs [hunk 1]
```diff
@@ -1,8 +1,14 @@
 use crate::core::prelude::*;
 
-pub fn archive_comments<D: Db>(db: &D, ids: &[&str]) -> Result<()> {
+pub fn archive_comments<D: Db>(db: &D, user_email: &str, ids: &[&str]) -> Result<()> {
     info!("Archiving {} comments", ids.len());
-    let archived = Timestamp::now();
-    db.archive_comments(ids, archived)?;
-    Ok(())
+    let users = db.get_users_by_email(user_email)?;
```
Comment (uklotzde) [inline src/core/usecases/archive_comments.rs:5]: TODO: Check Authorization for use cse by role -> higher level

File: src/core/usecases/archive_ratings.rs [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 
-pub fn archive_ratings<D: Db>(db: &D, ids: &[&str]) -> Result<()> {
+pub fn archive_ratings<D: Db>(db: &D, user_email: &str, ids: &[&str]) -> Result<()> {
     debug!("Archiving ratings {:?}", ids);
-    let archived = Timestamp::now();
-    db.archive_comments_of_ratings(ids, archived)?;
-    db.archive_ratings(ids, archived)?;
-    Ok(())
+    let users = db.get_users_by_email(user_email)?;
+    if let Some(user) = users.first() {
```
Comment (uklotzde) [inline src/core/usecases/archive_ratings.rs:6]: TODO: Check Authorization for use cse by role -> higher level

File: src/core/usecases/change_user_role.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+use crate::core::prelude::*;
+
+pub fn change_user_role<D: Db>(
+    db: &D,
+    account_email: &str,
```
Comment (uklotzde) [inline src/core/usecases/change_user_role.rs:5]: account_email = handelnder User

Comment (uklotzde) [reply src/core/usecases/change_user_role.rs:5]: = Authorization Context

File: src/core/usecases/archive_comments.rs [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+    if let Some(user) = users.first() {
```
Comment (uklotzde) [inline src/core/usecases/archive_comments.rs:6]: any

File: src/core/usecases/archive_ratings.rs (hunk 2 above)
Comment (uklotzde) [inline src/core/usecases/archive_ratings.rs:6]: any

File: src/core/usecases/change_user_role.rs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    user_email: &str,
+    role: Role,
+) -> Result<()> {
+    info!("Changing role to {:?} for {}", role, user_email);
+    let accounts = db.get_users_by_email(account_email)?;
+    let users = db.get_users_by_email(user_email)?;
+    let account = accounts
```
Comment (uklotzde) [inline src/core/usecases/change_user_role.rs:12]: any

File: src/infrastructure/flows/archive_ratings.rs [hunk 6]
```diff
@@ -2,12 +2,17 @@ use super::*;
 
 use diesel::connection::Connection;
 
-pub fn exec_archive_ratings(connections: &sqlite::Connections, ids: &[&str]) -> Result<()> {
+pub fn exec_archive_ratings(
+    connections: &sqlite::Connections,
+    account_email: &str,
+    ids: &[&str],
+) -> Result<()> {
+    //TODO: check if user is allowed to archive the ratings
```
Comment (uklotzde) [inline src/infrastructure/flows/archive_ratings.rs:10]: delete

**Comment by flosse** [comment]
It's already merged :)

**Pull request closed without merging**